homeassistant/components/caldav/__init__.py

```diff
@@ -29,6 +29,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up CalDAV from a config entry."""
     hass.data.setdefault(DOMAIN, {})
 
+    if entry.options.get(CONF_DAYS) is None:
+        hass.config_entries.async_update_entry(entry, options={CONF_DAYS: DEFAULT_DAYS})
```

Comment:
I think its reasonable to change the default in the future when not set, so perhaps not set this?

As an alternative: I am curious what you were planning to set the # of days? I was curious if we should just increase the default.  I saw this before at 7 and thought that it was pretty low. Not sure if it would solve your problem to increase the default and not offer this as an option.

Reply:
> I think its reasonable to change the default in the future when not set, so perhaps not set this?

this is just to add a value to the config entry options if there is no one, yet.

> As an alternative: I am curious what you were planning to set the # of days? I was curious if we should just increase the default. I saw this before at 7 and thought that it was pretty low. Not sure if it would solve your problem to increase the default and not offer this as an option.

The default for YAML config is [1 day](https://github.com/home-assistant/core/blob/4d75940cd21d183414c290c6c778bb75936a0763/homeassistant/components/caldav/calendar.py#L70) and changeable by the user on demand.
When configuring via UI, than we actually set 7 days which might be enough for many (_most_) of the users, but sometimes it is to less (_see #108003_).
So i thought to make it configurable again, would offer the most flexibility to the users.
My own use case is my birthdays calendar - i don't have much friends, therefor not much entries in there 🙈😁 so it happens that next date will be a few more days/weeks far

Reply:
Would there be a down side if we just set the default to 30 days? or 60 days?

Reply:
yeah, i think it might still not be enough for some users. tbh the link issue is the only one i know complaining about it, so i think the 7 days is already ok for most users. Maybe we should just mention the 7 days and the possibility to change it in the docs?

Reply:
one downside of statically increasing to 60 or 90 days, it might result in higher load on CalDAV servers, especial on well filled calendars 🤔

Reply:
If we can find a default that works for most folks (e.g. 14 days, 30 days) that seems like it be ideal to not need extra configuration options.

Reply:
please don't get me wrong, but why not give the user the possibility to change this on it's own preferences or needs?
Sure, we can also increase the default to 30 which might be also good for most users, but also might increase the load on their caldav servers (_since more events might be returned_) and and their HA instances (_since more events needs to be checked which is the very next one_).
Further i think most users are already satisfied with the 7 days, since we we got only the one issue till now.

Reply:
My general impression is we want to not offer additional configuration options unless needed, and not necessarily support use cases that are far outside the norm. If we can find a balance of end user simplicity and performance then that would be ideal. If its not possible to find that middle ground because its too expensive, then its not possible i guess.

(Perhaps bad idea: could also be adaptive (first try 7 days, then 14, then max 32 if no events are found).)

Reply:
@allenporter: I am the one who opened #108003 and I certainly would not call said garbage reminder once a month "far outside the norm". In fact, I initially was happy with the HA update that made the CalDAV integration available via GUI only to spend days wondering why my calendar did not work until I found out about this limitation of days which was nowhere documented.
Besides: just because I'm the only one who opened a bug report does not mean numerous others also fell for this limitation and silently just gave up and reverted to the manual YAML way.

Reply:
If you read my above responses I'm not against solving this use case.  I just don't want to add more configuration options. Instead you'd be surprised it didn't work and have to hunt down a config option you didn't know existed. We should just make it work for you out or the box with no additional action on your part.

I'm in favor of solving this in a few ways proposed above... either increasing the limit (is this really an efficiency problem? I'm skeptical) or doing something adaptive and more efficient. 

Happy to be overruled here by other code approvers, that's just my 2 cents.

Reply:
The logical solution would be the one initially supposed by @mib1185: an options dialogue letting the user decide how many days to check, maybe with a note that too many days might increase the server load (which of course depends on the infrastructure, overall number of calendars and individual entries to check and such). To top it: a separate days chooser for each calendar, maybe as simple as entities after the initial setup.
Anyway, it should be emphasized in the docs that only one single day is taken into account. Currently, it is only mentioned in the [manual configuration steps](https://www.home-assistant.io/integrations/caldav/) and very easily overlooked, especially if someone uses the GUI integration and thus will most likely never get the idea to check these manual steps. At least that is what I did or better did not before I opened what I believed to be a bug report.
I will happily update the current docs accordingly, but it seems more efficient to wait for the outcome of this proposed change to avoid the need to update the docs again shortly afterwards. :wink:

Reply:
I'm still on the point, that having this option does not overwhelm our users, because it is not part of the integration setup phase (_config flow_), but can later be changed by the user if needed (_options flow_).
Also, as mentioned before, I think the default setting of 7 days is suitable for most of our users, so only a few users will need this option and then have it. Last but not least we could describe the options in the docs (_eq. we do in [AVM Fritz!Tools](https://www.home-assistant.io/integrations/fritz#integration-options)_) - i'm happy to update the docs accordingly 🙂